phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixStatement.java

```diff
@@ -371,6 +482,23 @@ private PhoenixResultSet executeQuery(final CompilableStatement stmt,
                                 plan =
                                         connection.getQueryServices().getOptimizer()
                                                 .optimize(PhoenixStatement.this, plan);
+                                setLastQueryPlan(plan);
+
+                                //verify metadata for the table/view/index in the query plan
+                                if (plan.getTableRef() != null && validateLastDdlTimestamp) {
```

Comment:
Is it even possible to have `plan.getTableRef()` as null ?

Reply:
In case of `EXPLAIN <query>`, I had noticed that `plan.getTableRef()` was null.

Reply:
Lets add a comment so that it is clear to everyone.